crates/transaction-pool/Cargo.toml

```diff
@@ -61,7 +61,18 @@ proptest-arbitrary-interop = { workspace = true, optional = true }
 [dev-dependencies]
 reth-provider = { workspace = true, features = ["test-utils"] }
 reth-tracing.workspace = true
+reth-e2e-test-utils.workspace = true
+reth-node-ethereum.workspace = true
+reth-ethereum-engine-primitives.workspace = true
+reth-ethereum = { workspace = true, features = [
+    "test-utils",
+    "network",
+    "pool",
+    "provider",
+] }
```

Comment:
this we can't really do because this is a cyclic dep graph, that only works because this is a dev-dep, so we should flatten these.

or move this test into reth-ethereum-node entreily